sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/plugins/CosmosMetricsSparkPlugin.scala

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.spark.plugins
+
+import com.azure.cosmos.spark.{CosmosClientMetrics, CosmosConfigNames, CosmosConstants}
+import com.azure.cosmos.spark.diagnostics.BasicLoggingTrait
+import com.codahale.metrics.MetricRegistry
+import io.micrometer.core.instrument.MeterRegistry
+import org.apache.spark.SparkContext
+import org.apache.spark.api.plugin.{DriverPlugin, ExecutorPlugin, PluginContext, SparkPlugin}
+
+import java.util
+
+// scalastyle:off underscore.import
+import scala.collection.JavaConverters._
+// scalastyle:on underscore.import
+
+class CosmosMetricsSparkPlugin extends SparkPlugin with BasicLoggingTrait {
+  override def driverPlugin(): DriverPlugin = new CosmosMetricsSparkDriverPlugin()
+
+  override def executorPlugin(): ExecutorPlugin = new CosmosMetricsSparkExecutorPlugin()
+
+  private[this] def cleanupMeterRegistry(meterRegistry: Option[MeterRegistry]): Unit = {
+    meterRegistry match {
+      case Some(existingRegistry) => CosmosClientMetrics.removeMeterRegistry(existingRegistry)
+      case None =>
+    }
+  }
+
+  private class CosmosMetricsSparkDriverPlugin extends DriverPlugin with BasicLoggingTrait {
+    private[this] var slf4jReporterEnabled = CosmosConstants.defaultSlf4jMetricReporterEnabled
+    private[this] var metricsCollectionIntervalInSeconds = CosmosConstants.defaultMetricsIntervalInSeconds
+    private[this] var meterRegistry: Option[MeterRegistry] = None
+
+    override def init
+    (
+      sc: SparkContext,
+      pluginContext: PluginContext
+    ): java.util.Map[String, String] = {
+      val originalConfig = super.init(sc, pluginContext).asScala
+
+      slf4jReporterEnabled = sc
+        .getConf
+        .getBoolean(CosmosConfigNames.MetricsEnabledForSlf4j, defaultValue = true)
```

Comment:
should the default value using defaultSlf4jMetricReporterEnabled

Reply:
Fixed in next iteration